File: bankingOnDjango/models/PaymentCard.py
from django.db import models
from bankingOnDjango.models.CardType import CardType
from bankingOnDjango.models.CardStatus import CardStatus
from bankingOnDjango.models.CardNetwork import CardNetwork


# ======================================================================
# Class PaymentCard Declaration
# ======================================================================
class PaymentCard(models.Model):

    # ======================================================================
    # attribute declarations
    # ======================================================================
    card_number_value = models.CharField(max_length=200, null=True)
    embossed_name = models.CharField(max_length=200, null=True)
    expiry_month = models.IntegerField(null=True)
    expiry_year = models.IntegerField(null=True)
    bank = models.ForeignKey(
        "Bank", on_delete=models.CASCADE, null=True, blank=True, related_name="+"
    )
    account = models.ForeignKey(
        "Account", on_delete=models.CASCADE, null=True, blank=True, related_name="+"
    )
    customer = models.ForeignKey(
        "Customer", on_delete=models.CASCADE, null=True, blank=True, related_name="+"
    )
    transactions = models.ManyToManyField("Transaction", blank=True, related_name="+")
    card_type = models.CharField(
        max_length=64, null=True, choices=[(tag.name, tag.value) for tag in CardType]
    )
    card_status = models.CharField(
        max_length=64, null=True, choices=[(tag.name, tag.value) for tag in CardStatus]
    )
    network = models.CharField(
        max_length=64, null=True, choices=[(tag.name, tag.value) for tag in CardNetwork]
    )

    # ======================================================================
    # function declarations
    # ======================================================================
    def toString(self):
        str = ""
        str = str + self.value
        return str

    def __str__(self):
        return self.toString()

    def identity(self):
        return "PaymentCard"

    def objectType(self):
        return "PaymentCard"

File: bankingonaspdotnet/Domain/PaymentCard.cs

using bankingonaspdotnet.Contracts;

namespace bankingonaspdotnet.Domain;

public class PaymentCard
{
    public Guid Id { get; set; } = Guid.NewGuid();

 public virtual long? PaymentcardId { get; set; } 
 public virtual CardPAN? CardNumber { get; set; } 
 public virtual string? EmbossedName { get; set; } 
 public virtual int? ExpiryMonth { get; set; } 
 public virtual int? ExpiryYear { get; set; } 
public virtual Bank? Bank { get; set; } 
public virtual Account? Account { get; set; } 
public virtual Customer? Customer { get; set; } 
public virtual ICollection<Transaction> Transactions { get; set; } = new List<Transaction>();
 public virtual CardType? CardType { get; set; } 
 public virtual CardStatus? CardStatus { get; set; } 
 public virtual CardNetwork? Network { get; set; } 

    public static PaymentCard FromRequest(PaymentCardRequest request) {
        return new PaymentCard {
            Id = request.Id,
            CardNumber = request.CardNumber,
            EmbossedName = request.EmbossedName,
            ExpiryMonth = request.ExpiryMonth,
            ExpiryYear = request.ExpiryYear,
            CardType = request.CardType,
            CardStatus = request.CardStatus,
            Network = request.Network,
        };
    }
}
